### ISSUE TITLE
box plots are stacked with multiple x axes

### ISSUE THREAD

**Issue by ludis** [plotly/plotly.js#4508]
**Issue**
Boxplots are stacked when using multiple x-axes
**Repro**
https://codepen.io/ludis/pen/ExarvdY
**Expected behaviour**
Boxplots are drawn next to each other
**Actual behaviour**
Boxplots are drawn on top of each other
**Discussion**
If you comment out 'xaxis' in scatter 2 of the above codepen, box plots are drawn correctly

**Comment by etpinard** [comment]
I'm not entirely sure what your desired behaviour is.

By _next to each other_, do you mean the two boxes should be grouped about the same `x` position with the first trace to the left and the second trace to the right?

Or, I see that you defined a `var x = [10, 20]` in your codepen, so I'm presuming you want to draw one box at `x=10` and the other at `x=20`. If so, I think https://codepen.io/etpinard/pen/bGNyGXG has what you're looking for. The important part in that codepen is setting `xaxis2.matches: 'x'` in the layout to tell plotly.js to make both x-axes have the same range (and use trace data on x-axis `'x'` and x-axis `'x2'` to compute the autorange.

**Comment by ludis** [comment]
Thanks Étienne,
sorry for not being very clear. 
The main issue I'm having is that I want a boxplot consisting of quite a number of traces, with two different x axes, one on top and one at the bottom. Every trace in this boxplot should have it's own label, one at the top and one at the bottom. As soon as I tried that by overlaying the x axes, the traces got drawn on top of each other. `xaxis2.matches: 'x'` does the trick, however. It moves every trace to it's proper position.
I followed the example on [https://plot.ly/javascript/multiple-axes/] as closely as possible. Apparently there's no need there to use `matches` there. 
I have adjusted my codepen and now it works as expected.
Thanks again!
